tests/ga/test_regorus.py

```diff
@@ -58,42 +72,294 @@ def tearDownClass(cls):
+                json.dump(policy, policy_file, indent=4)
+                policy_file.flush()
+                engine = Regorus(policy_file.name, self.default_rule_path)
+                output = engine.eval_query(input_dict, "data.agent_extension_policy.extensions_to_download")
+                result = output['result'][0]['expressions'][0]['value']
+                download_allowed = result.get(ALLOWED_EXT).get("downloadAllowed")
+                self.assertTrue(download_allowed, msg="Extension is in allowlist, so downloadAllowed should be True.")
+
+    def test_download_allowed_should_depend_on_global_allowlist_rule_if_extension_not_in_allowlist(self):
+        """
+        If extension is not in the allowlist:
+         - downloadAllowed = true if global allowlist rule not enabled (allowlistOnly = false)
+         - downloadAllowed = false if global allowlist rule enabled (allowlistOnly = true)
+        """
+        allowlist_only_cases = [True, False]
+        for allowlist_only in allowlist_only_cases:
+            policy = {
+                "azureGuestAgentPolicy": {
+                    "signingRules": {
+                        "extensionSigned": False
+                    },
+                    "allowListOnly": allowlist_only
+                },
+                "azureGuestExtensionsPolicy": {
+                    ALLOWED_EXT: {}
+                }
+            }
+
+            input_dict = {
+                "extensions": {
+                    RANDOM_EXT: {}      # Not included in the allowlist.
+                }
+            }
+
+            with tempfile.NamedTemporaryFile(mode='w', suffix='.json', delete=True) as policy_file:
+                json.dump(policy, policy_file, indent=4)
+                policy_file.flush()
+                engine = Regorus(policy_file.name, self.default_rule_path)
+                output = engine.eval_query(input_dict, "data.agent_extension_policy.extensions_to_download")
+                result = output['result'][0]['expressions'][0]['value']
+                download_allowed = result.get(RANDOM_EXT).get("downloadAllowed")
+                self.assertEqual(download_allowed, (not allowlist_only),
+                                msg="Extension is not in allowlist, allowListOnly is {0} so downloadAllowed should be {1}"
+                                 .format(download_allowed, allowlist_only))
+
+    def test_signing_validated_should_depend_on_signature_if_individual_signing_rule_true(self):
+        """
+        If individual signing rule is enforced, signingValidated=true only if extension is signed.
+        Global signing rule shouldn't matter.
+        """
+        # Should be same result for global signing rule true/false, so we test both cases.
+        global_signing_rule_cases = [True, False]
+        for global_signing_rule in global_signing_rule_cases:
+            policy = {
+                "azureGuestAgentPolicy": {
```

Comment:
This is where I'm having confusion about the 'extensionSigned' property. 
In the policy, what does 'extensionSigned': True mean? I assume it means that the agent should enforce signature validation. 

In the input, what does signingRules.extensionSigned == True for an extension mean? I assume it means that the extension itself is signed. 

If this is the case then these two properties have different meanings and it would be more readable if they were named differently

If i'm completely misunderstanding this lmk

Reply:
there are three parts of the policy: rego, data, and input. Rego defines how to compare data and input. In the policy data we define rules as extensionSigned which means extension has to be signed or not. this is in the tag of "signing rules". In input agent provides the fact about the extension in the tag of "signingInfo" via extensionSigned flag. In rego we compare whether the user provided signingInfo matches the data defined signingRule.

Reply:
Does the property signing rules need to have the same name as the data in signing info? I think the rule should be named something like 'allowUnsignedExtensions' as Norberto suggested and the data should be named 'extensionSigned'

Reply:
Having the same name for the rule and data, when they convey different things is hard to follow in the code